@here/harp-mapview/lib/ElevationProvider.ts

```diff
@@ -0,0 +1,37 @@
+import { GeoCoordinates, TileKey, TilingScheme } from "@here/harp-geoutils";
+import { Vector2 } from "three";
+import { DisplacementMap } from "./DisplacementMap";
+
+export interface ElevationProvider {
+    /**
+     * Get elevation for a given geo point.
+     * @param geoPoint geo position to query height for.
+     * @param level Optional data level that should be used for getting the elevation.
+     *              If undefined the deepest available tile that contains the geoPoint will be
+     *              used.
+     * @returns The height at geoPoint or undefined if no tile was found that covers the geoPoint.
+     */
+    getHeight(geoPoint: GeoCoordinates, level?: number): number | undefined;
+
+    /**
+     * Cast a ray through the given screen position
+     * @param screenPoint screenPoint to cast through
+     * @returns GeoCoordinates of the intersection point with the surface or undefined if the
+     *          surface was not hit by the ray.
+     */
+    rayCast(screenPoint: Vector2): GeoCoordinates | undefined;
```

Comment:
How could this work without providing the camera position (lat, long, el)?

Reply:
Opps, I forgot to pass in the camera to the constructor of the TerrainElevationProvider, so it will just ask for the camera position and then construct the ray using its position etc.

Reply:
actually, I will pass in the MapView, otherwise I need to pass in the height / width of the canvas also (needed to convert the screen point to world coordinates).